src/Core/Content/Product/ProductException.php

```
@@ -113,4 +114,14 @@ public static function originalIdNotFound(string $originalId): self
             ['originalId' => $originalId]
         );
     }
+
+    public static function productNotFound(string $productId): self
+    {
+        return new self(
```

you need to throw the old exception here, and throw self in the next major